### PR TITLE
TST- Fixing issue with test_parquet test unexpectedly passing

### LINKED ISSUE #24479: TST: Test_parquet failing
#### Code Sample, a copy-pastable example if possible

```bash
pytest -x pandas
...
============================================= FAILURES =============================================
_____________________________________ test_cross_engine_pa_fp ______________________________________
[gw0] darwin -- Python 3.7.1 /Users/ruijing/anaconda3/envs/pandas-dev/bin/python
[XPASS(strict)] reading pa metadata failing on Windows/mac
...
=== 1 failed, 44359 passed, 1104 skipped, 962 xfailed, 16 xpassed, 76 warnings in 601.05 seconds ===

```

```python
@pytest.mark.xfail(is_platform_windows() or is_platform_mac(),
                   reason="reading pa metadata failing on Windows/mac")
def test_cross_engine_pa_fp(df_cross_compat, pa, fp):
```

#### Problem description

Building the development environment of pandas from latest master source, and then running the pytest suite on a fresh install with no changes, leads to `test_cross_engine_pa_fp` test in test_parquet.py, which is expected to fail, passing. This could be due to the fact that either Python 3.7 or OS X 10.14 now supports what this test is testing for, cross-compat with differing reading/writing engines. Since this is the case, loosening the strictness of this test, per @TomAugspurger suggestion, should fix this. I can open up a pr for this.

#### Expected Output

Unit tests should not report failures

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: 80295f9483176b1335226680f1b55f09ded9ad46
python: 3.7.1.final.0
python-bits: 64
OS: Darwin
OS-release: 18.2.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.0.dev0+1398.g80295f948
pytest: 4.0.2
pip: 18.1
setuptools: 40.6.3
Cython: 0.29.2
numpy: 1.15.4
scipy: 1.1.0
pyarrow: 0.11.1
xarray: 0.11.0
IPython: 7.2.0
sphinx: 1.8.2
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: 2.5.12
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.2
lxml.etree: 4.2.5
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.15
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: 0.2.0
fastparquet: 0.1.6
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by RjLi13**
- [x] closes #24479 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This pr aims to make the failing test_parquet not strict since it now passes at least on my machine of Mac 10.14 running the latest python 3.7 and conda.

### CODE REVIEW

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -201,7 +201,8 @@ def test_options_get_engine(fp, pa):
 
 
 @pytest.mark.xfail(is_platform_windows() or is_platform_mac(),
```
Comment (TomAugspurger) [inline pandas/tests/io/test_parquet.py]: Can you try removing the xfail entirely. It looks like it may be unnecessary now.

**Review by jreback — CHANGES_REQUESTED** [review]
add ``pyarrow`` and ``fastparquet`` to this build: https://github.com/pandas-dev/pandas/blob/master/ci/deps/azure-macos-35.yaml as these are not being tested. I suspect this will actually still fail. I think then this test needs to be conditionally skipped on older pyarrow / fastparquet versions. We could add *another* macosx build that is a bit more modern, or just relax the numpy on this build to have it install a more modern pyarrow. 

cc @datapythonista

**Comment by jreback** [comment]
@datapythonista do we have the xfails shows somewhere? e.g. https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=6035

**Comment by datapythonista** [comment]
I don't exactly understand the rationale behind the builds, but a modern Mac build sounds reasonable.

Without know much about the builds we have, what would make sense to me is to have a "base" dependencies file, and have a build with the same conda deps in Linux, Windows and Mac. Happy to work on this if it makes sense.

Regarding xfails, I wanted to standardize the output of all the builds in #23115 (which still needs a bit of work). Not sure what is the preferred output, I think we have `-r sxX` in some places (like the windows builds).

**Comment by jreback** [comment]
Seems to be passing now.

**Review by jreback — APPROVED** [review]

**Comment by RjLi13** [comment]
So to catch up a bit, we still need the xfail because of `pyarrow` and `fastparquet` having differing versions on windows? Or is it Mac now supports cross-compat with differing reading/writing engines, but Windows doesn't?

File: pandas/tests/io/test_parquet.py (hunk 1 above)
Comment (RjLi13) [reply pandas/tests/io/test_parquet.py]: I did do that in another branch and it passed the CI tests, but it seems there's discussion of pyarrow and fastparquet as well as windows causing problems? Not completely sure.

**Comment by jreback** [comment]
no i think this is ok now

**Comment by RjLi13** [comment]
@jreback I have an unpublished branch that removes the xfail completely from both windows and mac. It seems to be passing the CI tests. Can you advise if we should take out the xfail completely?

**Comment by jreback** [comment]
yeah might be good to try that

**Comment by RjLi13** [comment]
Looks like removing the xfail works, seems whatever issue arose from running this test (old dependency maybe?) is gone now.

**Comment by TomAugspurger** [comment]
Thanks @RjLi13!

**Pull request merged**